### PR TITLE
Add ChainerX test to test_allreduce_persistent.py

### PR DESCRIPTION
**Pull request by keisukefukuda**
This PR is a part of #8031. It adds ChainerX tests to `test_allreduce_persistent.py`.

### CODE REVIEW

File: tests/chainermn_tests/extensions_tests/test_allreduce_persistent.py [hunk 1]
```diff
@@ -20,7 +20,15 @@ def __init__(self, n_in=3, n_units=5, n_out=2):
...
 class TestAllreducePersistent(unittest.TestCase):
 
-    def _test(self, comm, model):
+    def _test(self, comm, model, use_gpu, use_chx):
+        if use_gpu:
+            # Use CuPy's Device class to force call cudaSetDevice()
+            chainer.cuda.get_device_from_id(comm.intra_rank).use()
+
+        device = get_device(comm.intra_rank if use_gpu else None, use_chx)
+        device.use()
```
Comment (belldandyxtq) [inline tests/chainermn_tests/extensions_tests/test_allreduce_persistent.py]: Unnecessary

File: tests/chainermn_tests/extensions_tests/test_allreduce_persistent.py [hunk 2]
```diff
@@ -39,14 +47,13 @@ def _test(self, comm, model):
 
     def test_allreduce_persistent_cpu(self):
         comm = chainermn.create_communicator('naive')
-        self._test(comm, ExampleModel())
+        model = ExampleModel()
+        self._test(comm, model, False, False)
```
Comment (belldandyxtq) [inline tests/chainermn_tests/extensions_tests/test_allreduce_persistent.py]: Can we have a comment here to make it clear for `chainerx` test and `numpy` test?

File: tests/chainermn_tests/extensions_tests/test_allreduce_persistent.py (hunk 1 above)
Comment (keisukefukuda) [reply tests/chainermn_tests/extensions_tests/test_allreduce_persistent.py]: I'm not sure this is unnecessary. CUDA's current GPU and ChainerX's current GPU are different. I'll consult ChainerX developers.

File: tests/chainermn_tests/extensions_tests/test_allreduce_persistent.py (hunk 2 above)
Comment (keisukefukuda) [reply tests/chainermn_tests/extensions_tests/test_allreduce_persistent.py]: Fixed.

File: tests/chainermn_tests/extensions_tests/test_allreduce_persistent.py (hunk 1 above)
Comment (belldandyxtq) [reply tests/chainermn_tests/extensions_tests/test_allreduce_persistent.py]: As we discussed, this code is unnecessary, the succeeding `to_device` serve the purpose.

Comment (keisukefukuda) [reply tests/chainermn_tests/extensions_tests/test_allreduce_persistent.py]: I'll fix it.

**Comment by belldandyxtq** [comment]
pfnci, test this please

**Comment by belldandyxtq** [comment]
The Error seems to be in the ChainerX test and not related to this PR, I will merge it

**Comment by belldandyxtq** [comment]
I rerun the execution, the error is clear now

**Pull request merged**